v2/pkg/engine/resolve/loader.go

```diff
@@ -1132,8 +975,14 @@ func (l *Loader) optionallyRewriteErrorPaths(fetchItem *FetchItem, values []*ast
 				newPath := make([]string, 0, len(pathPrefix)+len(pathItems)-i)
 				newPath = append(newPath, pathPrefix...)
 				for j := i + 1; j < len(pathItems); j++ {
-					// BUG: for pathItems containing integers, it will append empty strings
-					newPath = append(newPath, unsafebytes.BytesToString(pathItems[j].GetStringBytes()))
+					switch pathItems[j].Type() {
+					case astjson.TypeString:
+						newPath = append(newPath, unsafebytes.BytesToString(pathItems[j].GetStringBytes()))
+					case astjson.TypeNumber:
+						newPath = append(newPath, strconv.Itoa(pathItems[j].GetInt()))
```

Comment:
if we under "_entities" - I believe we need to skip next integer path item - because it will be index of concrete entity

right now we can't map it back to original location of object, so it will be more simpe to skip it